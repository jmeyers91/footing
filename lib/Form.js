"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importStar(require("react"));
var styled_components_1 = __importDefault(require("styled-components"));
function Form(props) {
    var onSubmit = props.onSubmit;
    var handleSubmit = react_1.useCallback(function (event) {
        event.preventDefault();
        var onSubmit = props.onSubmit;
        if (onSubmit) {
            onSubmit(event);
        }
    }, [onSubmit]);
    return react_1.default.createElement("form", __assign({}, props, { onSubmit: handleSubmit }));
}
exports.Form = Form;
/**
 * Form element that calls `preventDefault` on submit events automatically.
 * @param props
 */
exports.default = styled_components_1.default(Form)(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n  display: flex;\n  flex-direction: column;\n  flex-shrink: 0;\n  position: relative;\n  box-sizing: border-box;\n"], ["\n  display: flex;\n  flex-direction: column;\n  flex-shrink: 0;\n  position: relative;\n  box-sizing: border-box;\n"])));
var templateObject_1;
//# sourceMappingURL=Form.js.map