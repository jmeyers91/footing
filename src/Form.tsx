import React, { FormEvent, ComponentProps, useCallback } from 'react';
import styled from 'styled-components';

export function Form(props: ComponentProps<'form'>) {
  const { onSubmit } = props;

  const handleSubmit = useCallback(
    (event: FormEvent<HTMLFormElement>) => {
      event.preventDefault();
      const { onSubmit } = props;
      if (onSubmit) {
        onSubmit(event);
      }
    },
    [onSubmit]
  );

  return <form {...props} onSubmit={handleSubmit} />;
}

/**
 * Form element that calls `preventDefault` on submit events automatically.
 * @param props
 */
export default styled(Form)`
  display: flex;
  flex-direction: column;
  flex-shrink: 0;
  position: relative;
  box-sizing: border-box;
`;
