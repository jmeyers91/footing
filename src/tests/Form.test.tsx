import React from 'react';
import { render } from '@testing-library/react';
import Form from '../../src/Form';

describe('Form', () => {
  test('Should match snapshot', () => {
    const { container } = render(<Form />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
