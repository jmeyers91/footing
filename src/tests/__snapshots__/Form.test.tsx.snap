// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Should match snapshot 1`] = `
<form
  class="sc-bdVaJa eUEwfn"
/>
`;
